src/TableView.jl

```diff
@@ -1,119 +1,158 @@
 module TableView
-
-using WebIO
-using JSExpr
-using JuliaDB
-using DataValues
-
-import JuliaDB: DNDSparse, DNextTable, NextTable
-
-function JuliaDB.subtable(t::DNextTable, r)
-    table(collect(rows(t)[r]), pkey=t.pkey)
-end
-
-showna(xs) = xs
-function showna(xs::AbstractArray{T}) where {T<:DataValue}
-    map(xs) do x
-        isnull(x) ? "NA" : get(x)
+using Tables, TableTraits, IteratorInterfaceExtensions
+using WebIO, JSExpr, JSON, Dates
+using Observables: @map
+
+function showtable(x; dark = false)
+    if Tables.istable(typeof(x))
+        return _showtable(x, dark)
+    elseif TableTraits.isiterabletable(x)
```

Comment:
This also needs to handle the case where ``TableTraits.isiterabletable`` returns ``missing``, like the ``DataFrame`` constructor [here](https://github.com/JuliaData/DataFrames.jl/blob/ce9c24a539008cb0db646a6bdbf4e70ae5eab10d/src/other/tables.jl#L24).

Those cases are a bit tricky to generate right now [*] but they will become much more common with hopefully the next release of [Query.jl](https://github.com/queryverse/Query.jl). These are essentially cases where [Query.jl](https://github.com/queryverse/Query.jl) returns a lazy iterator but wasn't able to use type inference to determine the ``eltype`` of the iterator, and so a source needs to be able to handle that case. The general strategy in those cases is to look at the type of the first iterated element to figure out the schema. For iterable tables, there is a rule that they must iterate only elements of the same type, so looking at the first element should be enough to handle all iterable table sources.

[*] The shortest example right now is ``TableView.showtable(rand()>0.00000000001 ? i : true for i in df)``. I think there is a plan to handle to subsume generators under ``Tables.istable`` (that I disagree with), at which point this particular example would be moot. But once I've made the change in [Query.jl](https://github.com/queryverse/Query.jl) that I mentioned, this will show up independently of this example.

Reply:
Ah, I saw that case in the DataFrame constructor but couldn't figure out what it meant.

Isn't `isiterabletable` a bit of a misnomer it can return `missing` for something that most definitely is an iterable table, albeit with undetermined eltype?

Reply:
Well, in general there are cases where ``isiterabletable`` just can't figure out whether something is a table or not. A generator that returns a stream of ``Int``s is not a table, but it doesn't have an element type, so you can't tell until you start to iterate. So, of course a source either is or is not a table, but we don't know that when we call ``isiterabletable``, so ``missing`` seemed appropriate?

What is missing is the information whether something is a table or not.

Reply:
Fair enough, I guess -- I definitely haven't put as much thought into it as you :)